pkg/qemu/qemu.go

```diff
@@ -42,60 +102,77 @@ func NewNetwork() *Network {
 	}
 }
 
-func (n *Network) newVM() *networkState {
-	num := n.numVMs
-	n.numVMs++
-	return &networkState{
-		connect: num != 0,
-		mac:     net.HardwareAddr{0x0e, 0x00, 0x00, 0x00, 0x00, byte(num)},
-		port:    n.port,
+// Cmdline implements Device.
+func (n *Network) Cmdline() []string {
+	if n == nil {
+		return nil
 	}
-}
 
-type networkState struct {
-	// Whether to connect or listen.
-	connect bool
-	mac     net.HardwareAddr
-	port    uint16
-}
+	newNum := atomic.AddUint32(&n.numVMs, 1)
+	num := newNum - 1
+	mac := net.HardwareAddr{0x0e, 0x00, 0x00, 0x00, 0x00, byte(num)}
```

Comment:
What is this mac address? I suppose you just want a monotonically increasing address for the last byte. Why is there 0x0e for the first byte then? Is it defined by some spec? Would be nice to put a reference here.

Also this constrains us to 256 VMs on the network. Would be nice to either hard error out when you exceed 256 and print, or to extend capability for more than 256, since I think this might just cause silent weird errors if 2 VMs start sharing a mac address

Reply:
I just realized this was originally there. Do you mind at least putting a error or something?